doc/developers/contributing.rst

```diff
@@ -247,6 +247,9 @@ how to set up your git repository:
       git clone git@github.com:YourLogin/scikit-learn.git  # add --depth 1 if your connection is slow
       cd scikit-learn
 
+   In order to run the above clone command you should have an SSH key to connect to your GitHub account.
+   Please refer to `this guide <https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent>`_ if you hadn't set it yet.
```

Comment:
With the command we currently have in our doc, you do need SSH keys setup for cloning a public repo. For example this fails for me since I don't have SSH keys setup:
```
git clone git@github.com:scikit-learn/scikit-learn.git
```

Which one of the two sections would you advise, the `gh` CLI or the Git Credential Manager one?

Reply:
I noticed now that there's also a mismatch in the two documents here discussed. In `contributing.rst` the clone is performed using ssh, as pointed out by @lesteve:

https://github.com/scikit-learn/scikit-learn/blob/559533e4eb6925c3ce6cab47e77804e1cedddee0/doc/developers/contributing.rst?plain=1#L247-L248

while in `advanced_installation.rst`,  is made using git protocol:

https://github.com/scikit-learn/scikit-learn/blob/a4a53eaa223394204afa715dfb672b36dc8fbb40/doc/developers/advanced_installation.rst?plain=1#L50-L51

That also could run into issues if you are in a network restricted by a firewall. I think that it's better to rethink the documentation using the https connection, which is easier to manage

Reply:
To move this PR forward, lets revert this line about ssh keys so we can move forward with the other improvements.

We can work on authentication in a follo wup.

Reply:
Since when `git clone` was first introduced in the `advanced_installation.rst` through #10698 , it was included as it is now via the git clone protocol. So I will add a commit soon changing it to ssh in order to have consistency throughout these docs.

Reply:
Ok, got it